### ISSUE TITLE
Products created through the API do not have their media attached in the media library.

### ISSUE THREAD

**Issue by Babylon1999** [woothemes/woocommerce#37114]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When creating a new product using this API call `/wp-json/wc/v3/products`, the image `src` is added successfully but not attached to the product created.

![image](https://user-images.githubusercontent.com/67080558/223582238-0974253e-db7c-4387-95fd-96abc92c7ffb.png)

### Expected behavior

The expected behaviour is to attach the product ID as `post_parent` for any added image.

![image](https://user-images.githubusercontent.com/67080558/223582296-c8f22c9c-ed05-4826-8beb-c0de523bc94d.png)

### Actual behavior

The images are not attached.

### Steps to reproduce

1 - Configure WooCommerce API according to docs.

2- Create a new product with any framework, for example, Python:

```
from woocommerce import API

wcapi = API(
    url="https://xxx.wpcomstaging.com",
    consumer_key="xxxx",
    consumer_secret="xxxx",
    version="wc/v3"
)


data = {
    "name": "Premium Quality- API Test",
    "type": "simple",
    "regular_price": "21.99",
    "description": "Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.",
    "short_description": "Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.",
    "categories": [
        {
            "id": 9
        },
        {
            "id": 14
        }
    ],
    "images": [
        {
            "src": "http://demo.woothemes.com/woocommerce/wp-content/uploads/sites/56/2013/06/T_2_front.jpg"
        },
        {
            "src": "http://demo.woothemes.com/woocommerce/wp-content/uploads/sites/56/2013/06/T_2_back.jpg"
        }
    ]
}

print(wcapi.post("products", data).json())
```

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://xxxxxx.wpcomstaging.com
Site address (URL): https://xxxxxxxx.wpcomstaging.com
WC Version: 7.4.1
REST API Version: ✔ 7.4.1
WC Blocks Version: ✔ 9.4.4
Action Scheduler Version: ✔ 3.5.4
Log Directory Writable: ✔
WP Version: 6.1.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.0.28
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.86.0
OpenSSL/1.1.1n

`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by Babylon1999** [comment]
Reported in zd-6006340

**Comment by xue28** [comment]
6033253-zen

**Comment by rrennick** [comment]
Addressed in #40076.
